pkg/deploy/registry/deployconfig/etcd/etcd.go

```diff
@@ -10,17 +16,41 @@ import (
 	"k8s.io/kubernetes/pkg/storage"
 
 	"github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/deploy/api/validation"
 	"github.com/openshift/origin/pkg/deploy/registry/deployconfig"
+	"github.com/openshift/origin/pkg/deploy/util"
 )
 
 const DeploymentConfigPath string = "/deploymentconfigs"
 
+// DeploymentConfigStorage contains the REST storage information for both DeploymentConfigs
+// and their Scale subresources.
+type DeploymentConfigStorage struct {
+	DeploymentConfig *REST
+	Scale            *ScaleREST
```

Comment:
> ??

Oh, yuck.  Another anti-pattern from upstream?  Different things should be different.  We can deal with it later.